h2/src/main/org/h2/util/DateTimeUtils.java

```diff
@@ -877,6 +877,20 @@ private static SimpleDateFormat getDateFormat(String format, String locale,
         }
     }
 
+    /**
+     * Returns number of days in month.
+     *
+     * @param year the year
+     * @param month the month
+     * @return number of days in the specified month
+     */
+    public static int getDaysInMonth(int year, int month) {
+        if (month != 2) {
+            return NORMAL_DAYS_PER_MONTH[month];
+        }
+        return (year & 3) == 0 && (year < 1582 || year % 100 != 0 || year % 400 == 0) ? 29 : 28;
```

Comment:
please break this line up and add some comments as to what is going on, this stuff is super complicated and other people need to have some chance of figuring it out later on